scrape/scrape.go

```diff
@@ -1452,6 +1473,24 @@ loop:
 			}
 		}
 
+		// Update the scrape loop's interval and timeout from their respective labels.
+		if intervalLabelValue := lset.Get(model.ScrapeIntervalLabel); intervalLabelValue != "" {
```

Comment:
I'm just taking a very first glance here and have to run for now, but I noticed that something's being conflated here that probably shouldn't: this is looking at scraped metric labels, and not at target labels, right? As far as I understand, the issue was about being able to control these durations via target relabeling, not via metrics relabeling?

Reply:
Now reading that issue it doesn't make that 100% clear, but I'm pretty sure that's what it meant.

Reply:
Yes, the issue was to control via target relabeling, and I believe this still does. The `lset` here is generated in the `populateLabels` function, where other target labels such as `job` and `__metrics_path__` are set.

https://github.com/prometheus/prometheus/blob/693c948c593a40c5ede20c3ab9dc115717baaac3/scrape/target.go#L325-L430

They are mixed with metric labels, which is why `__scrape_interval__` and `__scrape_timeout__` are accessed like that. They're still target labels though, and if you look in the UI they show up in the target labels popup.

Reply:
:exploding_head: I guess you're right. Ok, I'll leave this to someone more into the current scrape code, sorry :laughing:

Though somehow I assumed that a new target would be created anyway in `scrapePool.sync()` because the hash would be different, but probably I'm missing more stuff without looking really closely.

Reply:
No worries it took me a little while to sort all this out too. I still don't totally know what's going on 🤣

Reply:
This is indeed not the correct place. The loop should have the correct interval and timeout directly, in reload and in create.

Reply:
Yup, I see my fault now. Sorry, Julius!